core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1787,6 +1795,22 @@ public long getNewBlockIdForFile(AlluxioURI path) throws FileDoesNotExistExcepti
     }
   }
 
+  /**
+   * In order to allow writing to read-only files when creating,
+   * we need to skip write permission check for files sometimes.
+   */
+  private boolean skipFileWritePermissionCheck(LockedInodePath inodePath)
```

Comment:
how to make sure this function is called only for a file ?  what if the dir is passed in?

Reply:
Added checking `inodePath.getInode().isFile()` in the method